From a34351f3e2c7970b900b26aeed3eed61f89a71a4 Mon Sep 17 00:00:00 2001
From: Claudia Pellegrino <claui@users.noreply.github.com>
Date: Sun, 23 Oct 2022 17:38:03 +0200
Subject: [PATCH] Fix `ValueError` in Python 3.8 or newer

Validation has become stricter in Python 3.8.
So we need to pass `validate=False` to the constructor in this one case,
lest Python will print the following error message:

> ValueError: Invalid format 'a really silly format that discards the actual message' for '%' style
---
 test/test_utils/test_message.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test_utils/test_message.py b/test/test_utils/test_message.py
index 60d052e..4c6cd58 100644
--- a/test/test_utils/test_message.py
+++ b/test/test_utils/test_message.py
@@ -23,7 +23,7 @@ def test_init(self, m_errormsg):
     def test_init_w_formatter(self, m_errormsg):
         logger = logging.getLogger('test_init_w_formatter')
         format_string = 'a really silly format that discards the actual message'
-        handler = message.UserMessageHandler(logging.Formatter(format_string))
+        handler = message.UserMessageHandler(logging.Formatter(format_string, validate=False))
         logger.addHandler(handler)
 
         logger.info("WEEEOOOWEEEOOOWEEEOOO")
